pre_commit/commands/run.py

```diff
@@ -61,7 +61,11 @@ def get_filenames(args, include_expr, exclude_expr):
             lambda: get_changed_files(args.origin, args.source),
         )
     elif args.files:
-        getter = git.get_files_matching(lambda: args.files)
+        relative_files = [
+            os.path.relpath(os.path.abspath(filename), git.get_root())
+            for filename in args.files
+        ]
```

Comment:
Agreed, that's what I did first, but then many of the tests would not include this logic: e.g. `tests/commands/run_test.py` uses the `Runner` class, and does not pass through `pre_commit.main`.
Hence my decision to put this code here, so it can be shared across more pieces of code.
What do you think ?